domain/transfer.go

```diff
@@ -87,13 +88,21 @@ func (TransferProcessHandler) HandleEvent(
 		r.FromAccountID = x.FromAccountID
 		r.ToAccountID = x.ToAccountID
 
-		s.ExecuteCommand(commands.DebitAccount{
-			TransactionID:   x.TransactionID,
-			AccountID:       x.FromAccountID,
-			TransactionType: messages.Transfer,
-			Amount:          x.Amount,
-			ScheduledDate:   x.ScheduledDate,
-		})
+		scheduled, err := time.Parse("2006-01-02", x.ScheduledDate)
```

Comment:
Perhaps this parsing could be moved to a method on the message, which panics if the date is invalid.
At the very least we should define a constant for the format alongside the messages.

Reply:
What do you mean moved to a method on the message? Do you mean function bind it to a message type, or put a function in the messages package?

Apparently we already have the format and helper functions defined (unexported) in the `domain` package at `domain/domain.go`. Should those be moved into the `messages` pacakge so they are available outside the domain?

Reply:
I have updated things to use the helper functions that were already defined. I can't use them in the `main.go` file though because they are not exported. What are your thoughts on that?